src/services/flag.service.ts

```diff
@@ -0,0 +1,29 @@
+import {Injectable} from '@angular/core';
+import {HttpClient} from '@angular/common/http';
+import {environment} from '../environments/environment';
+import {ServerMessage} from '../model/server-message';
+import {Listing} from "../model/listing";
+
+const COULD_NOT_CONNECT = 'Could not connect to server.';
+
+@Injectable()
+export class FlagService {
+
+  constructor(private http: HttpClient) {}
+
+  flagListing(listing: Listing, description: string, next?: (msg: ServerMessage) => void): void {
+    this.http.post<ServerMessage>(environment.serverUrl + '/flag-listing',
```

Comment:
Note: You will have to change the route here if you change it on the backend as requested in the backend PR.